### ISSUE TITLE
Empty diagnostics when using Python optimization

### ISSUE THREAD

**Issue by treiher** [AdaCore/langkit#485]
We have observed in RecordFlux that the diagnostics of our langkit-based parser is empty when `python -O` is used (https://github.com/Componolit/RecordFlux/issues/603). The root cause seems to be an assertion in the Python binding (`librflxlang/__init__.py`):

```Python
1120     @property
1121     def diagnostics(self):
1122         """
1123         Diagnostics for this unit.
1124         """
1125         count = _unit_diagnostic_count(self._c_value)
1126         result = []
1127         diag = Diagnostic._c_type()
1128         for i in range(count):
1129             assert _unit_diagnostic(self._c_value, i, ctypes.byref(diag))
1130             result.append(diag._wrap())
1131         return result
```

The function `_unit_diagnostic` seems to have side effects and is not executed when `-O` is used. If line 1129 is commented out, the issue also occurs without `-O`.

```Python
6313 _unit_diagnostic = _import_func(
6314     "rflx_unit_diagnostic",
6315     [AnalysisUnit._c_type, ctypes.c_uint, ctypes.POINTER(Diagnostic._c_type)],
6316     ctypes.c_int,
6317 )
```

We use `-O` for performance reasons, otherwise the assertions and contracts used in RecordFlux would make the execution significantly slower.

### Versions

GNAT: Community 2020
Langkit: 21.0.0

**Comment by pmderodat** [comment]
Hello @treiher,

Thank you for reporting this. A fix is pending review on #486.

**Comment by pmderodat** [comment]
Hello again,

This is now fixed in the master branch. Thanks again for reporting this bug!
